src/main/java/org/ibp/api/rest/labelprinting/GermplasmListLabelPrinting.java

```diff
@@ -286,4 +292,91 @@ private void getEntryDetailFieldDataRowValue(
 	void setMaxTotalResults(final int maxTotalResults) {
 		this.maxTotalResults = maxTotalResults;
 	}
+
+	@Override
+	public LabelPrintingPresetDTO getDefaultSetting(final LabelsInfoInput labelsInfoInput, final String programUUID) {
+		final List<GermplasmListMeasurementVariableDTO> germplasmListMeasurementVariableDTOs =
+			this.germplasmListDataService.getGermplasmListDataTableHeader(labelsInfoInput.getListId(), programUUID);
+		final ArrayList<Integer> fields = new ArrayList<>();
+		germplasmListMeasurementVariableDTOs.stream().forEach(dto -> {
+			final TermId termId = TermId.getById(dto.getTermId());
+
+			switch (termId) {
+				case GID_ACTIVE_LOTS_COUNT:
+					fields.add(LabelPrintingStaticField.LOTS.getFieldId());
+					break;
+				case GID_AVAILABLE_BALANCE:
+					fields.add(TermId.AVAILABLE_INVENTORY.getId());
+					break;
+				case GID_UNIT:
+					fields.add(TermId.UNITS_INVENTORY.getId());
+					break;
+				case IMMEDIATE_SOURCE_GID:
+					fields.add(LabelPrintingStaticField.INMEDIATE_SOURCE_GID.getFieldId());
+					break;
+				case IMMEDIATE_SOURCE_NAME:
+					fields.add(LabelPrintingStaticField.INMEDIATE_SOURCE_PREFERRED_NAME.getFieldId());
+					break;
+				case GROUP_SOURCE_GID:
+					fields.add(LabelPrintingStaticField.GROUP_SOURCE_GID.getFieldId());
+					break;
+				case GROUP_SOURCE_NAME:
+					fields.add(LabelPrintingStaticField.GROUP_SOURCE_PREFERRED_NAME.getFieldId());
+					break;
+				case GUID:
+					fields.add(LabelPrintingStaticField.GUID.getFieldId());
+					break;
+				case GERMPLASM_REFERENCE:
+					fields.add(LabelPrintingStaticField.REFERENCE.getFieldId());
+					break;
+				case CROSS:
+					fields.add(LabelPrintingStaticField.CROSS.getFieldId());
+					break;
+				case BREEDING_METHOD_NAME:
+					fields.add(TermId.BREEDING_METHOD.getId());
+					break;
+				case BREEDING_METHOD_ABBREVIATION:
+					fields.add(LabelPrintingStaticField.METHOD_CODE.getFieldId());
+					break;
+				case BREEDING_METHOD_GROUP:
+					fields.add(LabelPrintingStaticField.METHOD_GROUP.getFieldId());
+					break;
+				case DESIG:
+					fields.add(TermId.PREFERRED_NAME.getId());
+					break;
+				case FGID:
+					fields.add(TermId.CROSS_FEMALE_GID.getId());
+					break;
+				case MGID:
+					fields.add(TermId.CROSS_MALE_GID.getId());
+					break;
+				case MALE_PARENT:
+					fields.add(TermId.CROSS_MALE_PREFERRED_NAME.getId());
+					break;
+				case FEMALE_PARENT:
+					fields.add(TermId.CROSS_FEMALE_PREFERRED_NAME.getId());
+					break;
+				default:
+					if (dto.getTermId() == GermplasmListLabelPrinting.DRVNM_ID || dto.getTermId() > 10000) {
```

Comment:
Change GermplasmLabelPrinting#MAX_FIXED_TYPE_INDEX to protected and use that instead